### PR TITLE
Create basic batch ingest sample

### PR DESCRIPTION
**Pull request by chrisjs**
Create basic batch ingest sample

* Read from a file, apply processing, write to database
* Hook in Spring Cloud Task
* Ensure job can be registered and executed through data flow
* Create README documenting build steps and how to register / launch the job as a task via Data Flow shell

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@chrisjs Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-dataflow-samples&pullRequestId=36)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-cloud/spring-cloud-dataflow-samples&pullRequestId=36) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@chrisjs Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-dataflow-samples&pullRequestId=36)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-cloud/spring-cloud-dataflow-samples&pullRequestId=36) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@chrisjs Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-dataflow-samples&pullRequestId=36)!

File: batch/file-ingest/src/main/java/org/springframework/ingest/mapper/fieldset/PersonFieldSetMapper.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package org.springframework.ingest.mapper.fieldset;
+
+import org.springframework.ingest.domain.Person;
+import org.springframework.batch.item.file.mapping.FieldSetMapper;
+import org.springframework.batch.item.file.transform.FieldSet;
+
+public class PersonFieldSetMapper implements FieldSetMapper<Person> {
```
Comment (mminella) [inline batch/file-ingest/src/main/java/org/springframework/ingest/mapper/fieldset/PersonFieldSetMapper.java]: Given previous comments, I wouldn't expect this to need to remain.

File: batch/file-ingest/src/main/java/org/springframework/ingest/config/BatchConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	@Autowired
+	public BatchConfiguration(final DataSource dataSource, final JobBuilderFactory jobBuilderFactory,
+							  final StepBuilderFactory stepBuilderFactory) {
+		this.dataSource = dataSource;
+		this.jobBuilderFactory = jobBuilderFactory;
+		this.stepBuilderFactory = stepBuilderFactory;
+	}
+
+	@Bean
+	public ItemReader<Person> reader() {
```
[the remark this answers is not in the document]
Comment (chrisjs) [reply batch/file-ingest/src/main/java/org/springframework/ingest/config/BatchConfiguration.java]: sure - i only accounted for the case of doing the minimal sample to make things work as discussed. i can do this now or in the revisions for the next round - which is preferred?

File: batch/file-ingest/src/main/java/org/springframework/ingest/config/BatchConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+		FlatFileItemReader<Person> reader = new FlatFileItemReader<Person>();
+		reader.setResource(new ClassPathResource("data.csv"));
```
[the remark this answers is not in the document]
Comment (chrisjs) [reply batch/file-ingest/src/main/java/org/springframework/ingest/config/BatchConfiguration.java]: right; same deal here as with the ItemReader/StepScope comment above - let me know if this should be done now or later.

File: batch/file-ingest/src/main/java/org/springframework/ingest/config/BatchConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+		reader.setLineMapper(new DefaultLineMapper<Person>() {{
+			setLineTokenizer(new DelimitedLineTokenizer() {{
+				setNames(new String[] { "firstName", "lastName" });
+			}});
+
+			setFieldSetMapper(new PersonFieldSetMapper());
```
[the remark this answers is not in the document]
Comment (chrisjs) [reply batch/file-ingest/src/main/java/org/springframework/ingest/config/BatchConfiguration.java]: i prefer to make objects immutable - using BeanMapperFieldSetMapper would require setters on Person AFAIK

File: batch/file-ingest/src/main/java/org/springframework/ingest/mapper/fieldset/PersonFieldSetMapper.java (hunk 1 above)
Comment (chrisjs) [reply batch/file-ingest/src/main/java/org/springframework/ingest/mapper/fieldset/PersonFieldSetMapper.java]: per response; i would prefer making things immutable where possible but if this adds distraction to the sample i can remove it. let me know which you would prefer

File: batch/file-ingest/src/main/java/org/springframework/ingest/config/BatchConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+	public BatchConfiguration(final DataSource dataSource, final JobBuilderFactory jobBuilderFactory,
+							  final StepBuilderFactory stepBuilderFactory) {
+		this.dataSource = dataSource;
+		this.jobBuilderFactory = jobBuilderFactory;
+		this.stepBuilderFactory = stepBuilderFactory;
+	}
+
+	@Bean
+	public ItemReader<Person> reader() {
+		FlatFileItemReader<Person> reader = new FlatFileItemReader<Person>();
```
[the remark this answers is not in the document]
Comment (chrisjs) [reply batch/file-ingest/src/main/java/org/springframework/ingest/config/BatchConfiguration.java]: will update, that looks much cleaner 👍

File: batch/file-ingest/src/main/java/org/springframework/ingest/config/BatchConfiguration.java [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+	}
+
+	@Bean
+	public ItemProcessor<Person, Person> processor() {
+		return new PersonItemProcessor();
+	}
+
+	@Bean
+	public ItemWriter<Person> writer() {
+		JdbcBatchItemWriter<Person> writer = new JdbcBatchItemWriter<Person>();
```
[the remark this answers is not in the document]
Comment (chrisjs) [reply batch/file-ingest/src/main/java/org/springframework/ingest/config/BatchConfiguration.java]: will update, that looks much cleaner 👍

File: batch/file-ingest/src/main/java/org/springframework/ingest/config/BatchConfiguration.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (cppwfs) [reply batch/file-ingest/src/main/java/org/springframework/ingest/config/BatchConfiguration.java]: Now if possible.

[the remark this answers is not in the document]
Comment (cppwfs) [reply batch/file-ingest/src/main/java/org/springframework/ingest/config/BatchConfiguration.java]: i.e. as a part of this PR.

**Comment by chrisjs** [comment]
pr has been updated with requested changes

**Comment by dturanski** [comment]
Needs to be updated for the new project structure

**Comment by sabbyanandan** [comment]
@chrisjs: We have moved to a new project structure / UX / docs for the samples; it'd be great if you can adapt your PR to the new model. Sorry about that.

**Comment by chrisjs** [comment]
okay ill follow up with Dave - looks like the entire repo is document files except a jar

**Comment by cppwfs** [comment]
In this case I believe we want the code included in the project.   Not just the jar.  The goal is to show folks how to build a batch job that works with SCDF.

**Comment by chrisjs** [comment]
@cppwfs right; i was just referring to the only non adoc file in the repo which was - https://github.com/spring-cloud/spring-cloud-dataflow-samples/blob/master/src/main/asciidoc/tasks/simple-batch-job/batch-job-1.3.0.BUILD-SNAPSHOT.jar not bundling the batch sample code into a JAR

**Comment by cppwfs** [comment]
ahh. gotcha!

**Comment by chrisjs** [comment]
closing this PR in favor of https://github.com/spring-cloud/spring-cloud-dataflow-samples/pull/39 which reflects the updated project structure

**Pull request closed without merging**